spec/controllers/api/v1/keywords_controller_spec.rb

```diff
@@ -3,10 +3,79 @@
 require 'rails_helper'
 
 RSpec.describe API::V1::KeywordsController, type: :controller do
-  describe 'POST#create', api_authentication: :user do
+  describe 'GET#index' do
+    context 'given a request without params' do
+      it 'returns expected status' do
+        api_sign_in
+        get :index
+
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'returns expected meta data' do
+        user = api_sign_in
+        Fabricate.times(15, :keyword, user: user)
+        get :index
+        body = JSON.parse(response.body, symbolize_names: true)
+
+        expect(body[:meta]).to eq(page: 1, pages: 1, page_size: 20, records: 15)
+      end
+    end
+
+    context 'given a request with pagination params' do
+      context 'given a valid page number' do
+        it 'returns expected status' do
+          user = api_sign_in
+          Fabricate.times(4, :keyword, user: user)
+          get :index, params: { page: { number: 2, size: 2 } }
+
+          expect(response).to have_http_status(:success)
+        end
+
+        it 'returns expected meta data' do
+          user = api_sign_in
+          Fabricate.times(20, :keyword, user: user)
+          get :index, params: { page: { number: 2, size: 2 } }
+          body = JSON.parse(response.body, symbolize_names: true)
+
+          expect(body[:meta]).to eq(page: 2, pages: 10, page_size: 2, records: 20)
+        end
+      end
+
+      context 'given an invalid page number' do
+        it 'returns not_found status' do
+          api_sign_in
+          get :index, params: { page: { number: 1000, size: 2 } }
+
+          expect(response).to have_http_status(:not_found)
+        end
+      end
+    end
+
+    context 'given a request with search keyword params' do
+      it 'returns expected status' do
```

Comment:
For this case, we should test if it return a valid keyword

Reply:
Good catch! I forgot to add test cases to check about the response content, will add more test cases about this

Reply:
Fixed in [`ddb5fab` (#55)](https://github.com/suho/google-search-ruby/pull/55/commits/ddb5fabf379d63d5c7e6ad4edc49abbde8fe2297)